opendevin/controller/agent_controller.py

```diff
@@ -168,14 +171,23 @@ def get_agent_state(self):
     async def start_delegate(self, action: AgentDelegateAction):
         AgentCls: Type[Agent] = Agent.get_cls(action.agent)
         agent = AgentCls(llm=self.agent.llm)
+        state = State(
+            inputs=action.inputs or {},
+            iteration=self.state.iteration,
+            max_iterations=self.max_iterations,
```

Comment:
why are we setting `max_iterations` on both State and AgentController?

Reply:
This is the status quo but I’ll check why it was there the first place! It also confused me a bit.

Reply:
History:

1. `max_iterations` first appeared in `AgentController` - that makes sense to me.
2. You introduced `max_iterations` to `State` in https://github.com/OpenDevin/OpenDevin/pull/1760. I understand the reason - otherwise there's no way to get `max_iterations` in the agent itself.

I do feel like `max_iterations` should reside in `AgentController`. It's immutable and shouldn't really be part of `State` IMHO. We can create a new issue to track this... maybe some mini refactoring is needed.

Reply:
Got it, thanks! I must have forgotten to remove it from the controller.

I think adding it to State makes sense--State doesn't need to just be mutable state, it's meant to hold all the information that the agent is allowed to have access to.

Reply:
Haha sure. My database background makes me very conscious about memory overhead. Copying an integer around makes me sad, but man we are already working on python 😆 

I'll remove it from AgentController.